Source/Core/Core/Config/MainSettings.cpp

```diff
@@ -736,4 +737,13 @@ bool IsDefaultGCIFolderPathConfigured(ExpansionInterface::Slot slot)
 {
   return Config::Get(GetInfoForGCIPath(slot)).empty();
 }
+
+bool IsCheatsEnabled()
+{
+#ifdef USE_RETRO_ACHIEVEMENTS
+  if (Config::Get(Config::RA_HARDCORE_ENABLED))
+    Config::SetBaseOrCurrent(Config::MAIN_ENABLE_CHEATS, false);
+#endif  // USE_RETRO_ACHIEVEMENTS
+  return Config::Get(Config::MAIN_ENABLE_CHEATS);
```

Comment:
This also seems like it would be better expressed as `Config::Get(Config::MAIN_ENABLE_CHEATS) && !Config::Get(Config::RA_HARDCORE_ENABLED)`